design/ceph/object/store.md

```diff
@@ -119,3 +119,46 @@ Specifying this section also ensures that the pool section in the ceph-object-zo
   zone:
     name: "name"
 ```
+
+### LDAP
+
+The Ceph Object Gateway supports integrating with LDAP for authenticating and creating users, please refer [here](https://docs.ceph.com/docs/master/radosgw/ldap-auth/). This means that the `rgw backend user` is also required to be part of groups in the LDAP server otherwise, authentication will fail. The `rgw backend user` can be generated from `CephObjectStoreUser` or `ObjectBucketClaim` CRDs. For the both resources credentials are saved in Kubernetes Secrets which may not be valid with `LDAP Server`, user need to follow the steps mentioned [here](https://docs.ceph.com/en/latest/radosgw/ldap-auth/#generating-an-access-token-for-ldap-authentication).The following settings need to be configured in the RGW server:
+```
+rgw ldap binddn =
+rgw ldap secret = /etc/ceph/ldap/bindpass.secret
+rgw ldap uri =
+rgw ldap searchdn =
+rgw ldap dnattr =
+rgw ldap searchfilter =
+rgw s3 auth use ldap = true
+```
+So the CRD for the Ceph Object Store will be modified to include the above changes:
+```yaml
+spec:
+  security
+    ldap:
+      config:
```

Comment:
Could we just include the settings directly under `ldap` instead of adding another `config` element? Everything in the CRDs is conceptually config
```yaml
spec:
  security:
    ldap:
      uri: ldaps://ldap-server:636
      binddn: "uid=ceph,cn=users,cn=accounts,dc=example,dc=com"
      searchdn: "cn=users,cn=accounts,dc=example,dc=com"
      dnattr: "uid"
      searchfilter: "memberof=cn=s3,cn=groups,cn=accounts,dc=example,dc=com"
      credential:
        secret: object-my-store-ldap-creds
```

Reply:
Or was there a need to define `config` as a generic `map[string]string` to allow for arbitrary settings to be passed to the config?

Reply:
I agree with @travisn... `config` is a terrible name for a nested map.

Reply:
As per discussions made, sticking with the same options than the string map